lib/pbench/test/unit/server/query_apis/test_namespace_and_rows.py

```diff
@@ -22,13 +23,65 @@ class TestIterationSamplesNamespace(Commons):
     @pytest.fixture(autouse=True)
     def _setup(self, client):
         super()._setup(
-            cls_obj=IterationSampleNamespace(client.config, client.logger),
-            pbench_endpoint="/dataset/samples/namespace",
+            cls_obj=SampleNamespace(client.config, client.logger),
+            pbench_endpoint="/index/namespace/iterations",
             elastic_endpoint="/_search",
             payload={"run_id": "random_md5_string1"},
             use_index_from_metadata=True,
         )
 
+    def test_with_no_index_document(self, client, server_config):
+        """
+        Check the Namespace API when no index name is provided
+        """
+        with client:
+            response = client.get(f"{server_config.rest_uri}/dataset/samples/namespace")
+            assert response.status_code == HTTPStatus.NOT_FOUND
+
+    def test_with_incorrect_index_document(self, provide_metadata):
+        """
+        Check the Namespace API when an incorrect index name is provided.
+        currently we only support iterations (result-data-samples) and
+        timeseries (result-data) documents
+        """
+        drb = Dataset.attach(controller="node", name="drb")
+
+        # When target_document encoded in URI is `test` we expect 404
+        with pytest.raises(NotFound) as exc:
+            self.cls_obj.assemble(
+                json_data={"target_document": "test"},
+                context={"run_id": "random_md5_string1", "dataset": drb},
+            )
+        assert exc.value.code == HTTPStatus.NOT_FOUND
+
+    def test_get_aggregatable_fields(self, attach_dataset, find_template):
+        template = Template.find("result-data")
+        mappings = {
+            "properties": {
+                key: value
+                for key, value in template.mappings["properties"].items()
+                if key in self.cls_obj.WHITELIST_AGGS_FIELDS["result-data"]
+            }
+        }
+        result = self.cls_obj.get_aggregatable_fields(mappings)
+        assert result == [
+            "@timestamp",
+            "@timestamp_original",
+            "result_data_sample_parent",
+            "run.id",
+            "run.name",
+            "iteration.name",
+            "iteration.number",
+            "sample.@idx",
+            "sample.name",
+            "sample.measurement_type",
+            "sample.measurement_idx",
+            "sample.uid",
+            "result.@idx",
+            "result.read_or_write",
+            "result.value",
+        ]
```

Comment:
I feel like this is too literal or realistic to be a good unit test.  E.g., `get_aggregatable_fields()` is completely orthogonal to `WHITELIST_AGGS_FIELDS`, so there should be no need to reference the latter in this test.  The whitelist is only being used because you're starting with a mapping from `Template.find("result-data")`, which is way more ornate than I think this test needs.  And, because the mapping comes from "far away" it's hard for me as a reviewer to tell whether the assertion which checks `result` is correct.

I think this test has only few cases to cover:
 - nested mappings (ideally, at least three levels deep)
 - mappings with `"field"` entries (ideally, at all three levels)
 - mappings which have type `"text"` (ideally, at all three levels)
 - one or more mappings which have a type other than `"text"` (ideally, at all three levels)

If you hand-craft an appropriate mapping dict here in this function, then the assertion should be able to cover everything with a single result, and it will be straightforward for the reviewer to match up the pieces.